SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -54,3 +54,43 @@ extension Array {
         }
     }
 }
+
+extension CollectionType {
+    public var ip_middleValues: [Generator.Element]? {
+        guard count > 0 else { return nil }
+        let needsAverageOfTwo = count.toIntMax().ip_isEven
+        let middle = startIndex.advancedBy(count / 2)
+        if needsAverageOfTwo {
+            let leftOfMiddle = startIndex.advancedBy((count / 2) - 1)
+            return [self[middle], self[leftOfMiddle]]
+        } else {
+            return [self[middle]]
+        }
+    }
+}
+
+extension SequenceType where Generator.Element: Equatable {
+    public func ip_mostCommonElements() -> [Generator.Element]? {
```

Comment:
That would return the original array if they all have a count of 1, actually. Some of the average functions are dependent on this, so I say let's figure it out now.